### PR TITLE
Add forms for works. Fixes #375

### LINKED ISSUE #375: Don't permit all attributes for curation_concerns
https://github.com/projecthydra-labs/curation_concerns/blob/master/app/controllers/concerns/curation_concerns/curation_concern_controller.rb#L140

This could enable any user who can edit an object to set an attribute they are not authorized to set. (i.e. Admin Collection)

### PR DESCRIPTION
**Pull request by jcoyne**

### CODE REVIEW

File: app/forms/curation_concerns/forms/work_form.rb [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+                    :visibility]
+
+      # @param [ActiveFedora::Base,#member_ids] model
+      # @param [Ability] current_ability
+      def initialize(model, current_ability)
+        @model = model
+        @current_ability = current_ability
+      end
+
+      # Uused to populate the representative dropdown
```
Comment (mjgiarlo) [inline app/forms/curation_concerns/forms/work_form.rb]: s/Uused/Used/

File: app/forms/curation_concerns/forms/work_form.rb [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+      # @return [Hash] All file sets in the collection, file.to_s is the key, file.id is the value
+      def files_hash
+        Hash[file_presenters.map { |file| [file.to_s, file.id] }]
+      end
+
+      private
+
+        # @return [Array<FileSetPresenter>] presenters for the file sets in order of the ids
+        def file_presenters
+          @file_sets ||= begin
```
Comment (mjgiarlo) [inline app/forms/curation_concerns/forms/work_form.rb]: Why the `begin` block?

File: lib/generators/curation_concerns/work/templates/form_spec.rb.erb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+# Generated via
+#  `rails generate curation_concerns:work <%= class_name %>`
+require 'rails_helper'
+
+describe CurationConcerns::<%= class_name %>Form do
+  it "should have some tests" do
```
Comment (mjgiarlo) [inline lib/generators/curation_concerns/work/templates/form_spec.rb.erb]: Could we `s/should have/has/` here to encourage writing tests that conform to our style?

File: spec/controllers/curation_concerns/generic_works_controller_json_spec.rb [hunk 4]
```diff
@@ -66,10 +73,12 @@
         expect(response.location).to eq main_app.curation_concerns_generic_work_path(created_resource)
       end
     end
+
     describe 'failed update' do
-      before { post :update, id: resource, generic_work: { title: nil, depositor: nil }, format: :json }
+      before { post :update, id: resource, generic_work: { title: [] }, format: :json }
```
Comment (mjgiarlo) [inline spec/controllers/curation_concerns/generic_works_controller_json_spec.rb]: Can you comment on this change? Not clear why it's needed/related.

File: app/forms/curation_concerns/forms/work_form.rb (hunk 2 above)
Comment (jcoyne) [inline app/forms/curation_concerns/forms/work_form.rb]: I'm just moving this from the previous location at `app/presenters/curation_concerns/form_presenter.rb`. There's no reason for it. I'll remove it.

File: spec/controllers/curation_concerns/generic_works_controller_json_spec.rb (hunk 4 above)
Comment (jcoyne) [inline spec/controllers/curation_concerns/generic_works_controller_json_spec.rb]: The whole point of this change was to prohibit certain unpermitted attributes (like depositor) from being passed as parameters and set automatically.

File: lib/generators/curation_concerns/work/templates/form_spec.rb.erb (hunk 3 above)
Comment (jcoyne) [inline lib/generators/curation_concerns/work/templates/form_spec.rb.erb]: let's do that as a new PR that changes all the generated spec files then.

File: spec/controllers/curation_concerns/generic_works_controller_json_spec.rb (hunk 4 above)
Comment (mjgiarlo) [inline spec/controllers/curation_concerns/generic_works_controller_json_spec.rb]: OK, is there a test that makes sure we can't pass depositor, or is our coverage already sufficient to make sure the form does as expected?

File: lib/generators/curation_concerns/work/templates/form_spec.rb.erb (hunk 3 above)
Comment (mjgiarlo) [inline lib/generators/curation_concerns/work/templates/form_spec.rb.erb]: :+1:

File: spec/controllers/curation_concerns/generic_works_controller_json_spec.rb (hunk 4 above)
Comment (jcoyne) [inline spec/controllers/curation_concerns/generic_works_controller_json_spec.rb]: I think the tests in hydra-derivatives and in `spec/forms/work_form_spec.rb` cover this.

**Pull request merged**